cookbooks/finite_strain/finite_strain.cc

```diff
@@ -80,16 +80,17 @@ namespace aspect
           for (unsigned int q=0; q < in.n_evaluation_points(); ++q)
             {
               // Convert the compositional fields into the tensor quantity they represent.
-              Tensor<2,dim> strain;
-              for (unsigned int i = 0; i < Tensor<2,dim>::n_independent_components ; ++i)
-                strain[Tensor<2,dim>::unrolled_to_component_indices(i)] = in.composition[q][i];
+              const ArrayView<const double> strain_values(&in.composition[q][0],
+                                                          Tensor<2,dim>::n_independent_components);
+              const Tensor<2,dim> strain(strain_values);
 
               // Compute the strain accumulated in this timestep.
               const Tensor<2,dim> strain_increment = this->get_timestep() * (velocity_gradients[q] * strain);
 
               // Output the strain increment component-wise to its respective compositional field's reaction terms.
-              for (unsigned int i = 0; i < Tensor<2,dim>::n_independent_components ; ++i)
-                out.reaction_terms[q][i] = strain_increment[Tensor<2,dim>::unrolled_to_component_indices(i)];
+              const ArrayView<double> strain_increment_values(&out.reaction_terms[q][0],
+                                                              Tensor<2,dim>::n_independent_components);
+              strain_increment.unroll(strain_increment_values.begin(), strain_increment_values.end());
```

Comment:
I don't think this is easier to read. The issue with `unroll()` is that it isn't clear which direction the operation goes. Is the tensor putting itself into an array, or is it reading new values from the array? This was clear in the original code.